app/server/lib/DocApi.ts

```diff
@@ -1734,11 +1733,20 @@ export class DocWorkerApi {
     return docAuth.docId!;
   }
 
-  private _getDownloadOptions(req: Request, name: string): DownloadOptions {
+  private async _getDownloadFilename(req: Request, tableId?: string): Promise<string> {
+    // Query DB for doc metadata to get the doc data.
+    const doc = await this._dbManager.getDoc(req);
+    const docTitle = doc.name;
+    const sufix = tableId ? (tableId === docTitle ? '' : `_${tableId}`) : '';
```

Comment:
- Nitpicking: `suffix` is the correct spelling;
 - Previously, the code used a `-` to separate the table and the doc names, you seem to change the separator for `_`, is there any reason for that change?